File: languages/php/Bloock/Entity/Availability/IpnsKey.php
<?php

namespace Bloock\Entity\Availability;

use Exception;

/**
 * Represents an object with a key uuid identifier.
 */
class IpnsKey
{
    public string $keyID;

    /**
     * Creates an IpnsKey instance with a key uuid identifier.
     * @param $keyID
     * @throws Exception
     */
    public function __construct($keyID)
    {
        $this->keyID = $keyID;
    }

    public function toProto(): \Bloock\IpnsKey
    {
        $s = new \Bloock\IpnsKey();
        $s->setKeyId($this->keyID);
        return $s;
    }

    public static function fromProto(?\Bloock\IpnsKey $ipnsKey = null): ?IpnsKey
    {
        if ($ipnsKey != null) {
            return new IpnsKey($ipnsKey->getKeyId());
        }
        return null;       
    }
}

File: languages/python/bloock/entity/availability/ipns_key.py
from __future__ import annotations

import bloock._bridge.proto.bloock_availability_entities_pb2 as proto


class IpnsKey:
    """
    Represents an object with a key uuid identifier.
    """
    key_id = None

    def __init__(self, key_id: str) -> None:
        """
        Creates an IpnsKey instance with a key uuid identifier.
        :type key_id: object
        :rtype: object
        """
        self.key_id = key_id

    def to_proto(self) -> proto.IpnsKey:
        return proto.IpnsKey(
            key_id=self.key_id
        )
    
    @staticmethod
    def from_proto(res: proto.IpnsKey = None) -> IpnsKey:
        if isinstance(res, proto.IpnsKey):
            return IpnsKey(res.key_id)
        return None